### PR TITLE
More immutable profile

### PR DESCRIPTION
**Pull request by VilppeRiskidev**

### CODE REVIEW

**Comment by VilppeRiskidev** [comment]
This kind of is two things in one, would've been better to split into two (or even more...) separate PRs. Upcoming PRs will be kept smaller.

File: src/r2mm/installing/ConflictManagementProviderImpl.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export default class ConflictManagementProviderImpl extends ConflictManagementPr
     }
 
     // TODO: Iterate through mods array and look for anything out of place.
-    async resolveConflicts(mods: ManifestV2[], profile: Profile): Promise<R2Error | void> {
+    async resolveConflicts(mods: ManifestV2[], profile: ImmutableProfile): Promise<R2Error | void> {
```
Comment (anttimaki) [inline src/r2mm/installing/ConflictManagementProviderImpl.ts:42]: Minor: clean up `Profile` from imports.

**Review by anttimaki — CHANGES_REQUESTED** [review]
Requires a small bug fix, otherwise looks good!

**Review by anttimaki — APPROVED** [review]
Nice

**Pull request merged**